oars/macros/utils.sql

```diff
@@ -0,0 +1,89 @@
+{% macro extract_video_interacted_type(event_str) %}
+  case
+    when JSON_EXISTS(event_str, '$.result.extensions."https://w3id.org/xapi/video/extensions/cc-enabled"')
+      -- as far as I can tell, there's no way to differentiate between transcript shown/hidden
+      -- and closed captioning shown/hidden, so the event name is chosen to reflect that
+      -- it can refer to either interaction
+      then 'video_cc_transcript_toggled'
+    when JSON_EXISTS(event_str, '$.result.extensions."https://w3id.org/xapi/openedx/extension/speed-to"')
+      then 'video_speed_changed'
+    else ''
+  end
+{% endmacro %}
+
+
+-- given an event object string, return the event type (e.g. video_loaded)
+{% macro get_event_type(event_str, verb_id) %}
+  case
+    when {{ verb_id }} like 'https://w3id.org/xapi/video/verbs/paused'
+      then 'edx.video.paused'
```

Comment:
I'd love to avoid having any tracking log names in here if we can avoid it. Could we use the display names included in the xapi `verb.display` here?

Reply:
Sure thing. Would xAPI events always have `en` as the key in `verb.display`, or would this key need to be parameterized at some point?

Reply:
It's hardcoded to "en" right now on the sending side, so I think we can go forward assuming that will be the case for a while.